python/pyspark/sql/connect/client.py

```diff
@@ -182,6 +191,10 @@ def _to_pandas(self, plan: pb2.Plan) -> Optional[pandas.DataFrame]:
         req = pb2.Request()
         req.user_context.user_id = self._user_id
         req.plan.CopyFrom(plan)
+        if self.has_arrow:
+            req.preferred_result_type = pb2.Request.ArrowBatch
+        else:
+            req.preferred_result_type = pb2.Request.JSONBatch
```

Comment:
We could remove this and just let it throw an exception. I think we can just make the Arrow as a hard requirement. Since this is a new code not released yet, let's go with a stricter approach. cc @grundprinzip FYI.

We can remove this in a separate PR too.

Reply:
i notice that pyspark checks whether schema is supported https://github.com/apache/spark/blob/master/python/pyspark/sql/pandas/conversion.py#L102 ,
i move this check to the server side since it needs schema.

maybe we can keep it as a fallback if json could support more data types (not sure about this)